crates/client/mempool/src/block_production.rs

```diff
@@ -287,35 +326,53 @@ impl BlockProductionTask {
 
         let n_pending_ticks_per_block = self.backend.chain_config().n_pending_ticks_per_block();
 
-        log::debug!("begin pending tick {}/{}", current_pending_tick, n_pending_ticks_per_block);
+        if current_pending_tick == 0 {
+            return Ok(());
+        }
 
         // Reduced bouncer capacity for the current pending tick
 
-        let config_bouncer = self.executor.bouncer.bouncer_config.block_max_capacity;
-        let frac = n_pending_ticks_per_block / current_pending_tick; // div by zero: current_pending_tick has been checked for 0 above
+        let config_bouncer = self.backend.chain_config().bouncer_config.block_max_capacity;
+
+        // reduced_gas = gas * current_pending_tick/n_pending_ticks_per_block
+        // - we're dealing with integers here so prefer having the division last
+        // - use u128 here because the multiplication would overflow
+        // - div by zero: current_pending_tick has been checked for 0 above
+        let reduced_cap =
+            |v: usize| (v as u128 * current_pending_tick as u128 / n_pending_ticks_per_block as u128) as usize;
```

Comment:
`n_pending_ticks_per_block` should be checked instead of `current_pending_tick`!

Reply:
:bangbang: